client/src/pages/ViewRequestStatus/RequestStatusView.js

```diff
@@ -1,20 +1,77 @@
 import React, { useEffect, useState } from "react";
+// import { Navigate } from "react-router-dom";
 
 import { useParams } from "react-router-dom";
 import ButtonComponent from "./ButtonComponent";
 
 export function RequestStatus() {
 	const [request, setRequest] = useState(null);
+	const [donation, setDonation] = useState(null);
+	const [isCancelled, setIsCancelled] = useState(false);
```

Comment:
probably you just needed a needsReloading state, along with a setNeedsReloading, but this is fine too

Reply:
(instead of different states for the isCancelled, isRejected, etc transition)

Reply:
Done that